File: First Semester/FP - Fundamentals of programming/FinalProject/lab79/domain/Entity.py

class Entity:

    def __init__(self, id):
        self.__id = id

    def get_id(self):
        return self.__id

    def set_id(self, new_id):
        self.__id = new_id
        
    def __str__(self):
        return "ID" + str(self.get_id()) + "\n"

File: Fourth Semester/Software Engineering/src/main/java/com/example/bibliotecaiss/domain/Entity.java
package com.example.bibliotecaiss.domain;

import java.io.Serializable;

public class Entity<ID> implements Serializable {
    protected ID id;
    public ID getId() {
        return id;
    }
    public void setId(ID id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Entity)) return false;
        Entity<?> entity = (Entity<?>) o;
        return getId().equals(entity.getId());
    }

    @Override
    public int hashCode() {
        return java.util.Objects.hash(getId());
    }

    @Override
    public String toString() {
        return "Entity{" +
                "id=" + id +
                '}';
    }
}
